src/Mod/Sketcher/App/Sketch.cpp

```diff
@@ -2691,7 +2690,42 @@ int Sketch::addAngleAtPointConstraint(
     if(avp)
         tag = ++ConstraintsCounter;
 
-    GCSsys.addConstraintAngleViaPoint(*crv1, *crv2, p, angle, tag, driving);
+    if (Geoms[geoId1].type == BSpline || Geoms[geoId2].type == BSpline) {
+        if (Geoms[geoId1].type == BSpline && Geoms[geoId2].type == BSpline) {
+            GCS::Point &p3 = Points[getPointId(geoId3, pos3)];
+            auto* partBsp = static_cast<GeomBSplineCurve*>(Geoms[geoId1].geo);
+            double uNear;
+            partBsp->closestParameter(Base::Vector3d(*p3.x, *p3.y, 0.0), uNear);
+            double* pointparam1 = new double(uNear);
+            Parameters.push_back(pointparam1);
+            addPointOnObjectConstraint(geoId3, pos3, geoId1, pointparam1, driving); //increases ConstraintsCounter
+            --ConstraintsCounter;
+            partBsp = static_cast<GeomBSplineCurve*>(Geoms[geoId2].geo);
+            partBsp->closestParameter(Base::Vector3d(*p3.x, *p3.y, 0.0), uNear);
+            double* pointparam2 = new double(uNear);
+            addPointOnObjectConstraint(geoId3, pos3, geoId2, pointparam2, driving); //increases ConstraintsCounter
+            Parameters.push_back(pointparam2);
+            --ConstraintsCounter;
+            GCSsys.addConstraintAngleViaPointAndTwoParams(*crv1, *crv2, p, pointparam1, pointparam2, angle, tag, driving);
+        }
+        else {
+            if (Geoms[geoId1].type != BSpline) {
+                std::swap(geoId1, geoId2);
+                std::swap(pos1, pos2);
+            }
+            GCS::Point &p3 = Points[getPointId(geoId3, pos3)];
+            auto* partBsp = static_cast<GeomBSplineCurve*>(Geoms[geoId1].geo);
+            double uNear;
+            partBsp->closestParameter(Base::Vector3d(*p3.x, *p3.y, 0.0), uNear);
+            double* pointparam = new double(uNear);
+            Parameters.push_back(pointparam);
+            int tag = addPointOnObjectConstraint(geoId3, pos3, geoId1, pointparam, driving); //increases ConstraintsCounter
+            --ConstraintsCounter;
+            GCSsys.addConstraintAngleViaPointAndParam(*crv1, *crv2, p, pointparam, angle, tag, driving);
+        }
+    }
+    else
+        GCSsys.addConstraintAngleViaPoint(*crv1, *crv2, p, angle, tag, driving);

```

Comment:
This will move to the method that calls this method. This may mean having to split this method as well.